tests/unit_tests/crud/test_stats.py

```diff
@@ -0,0 +1,233 @@
+from unittest.mock import MagicMock
+
+import pytest
+from elasticsearch_dsl.response import Response
+
+from app.crud.elastic import ResourceType
+from app.crud.stats import score, score_search
+from app.score import ScoreModulator, ScoreWeights, calc_scores, calc_weighted_score
+
+
+def test_score_empty_hits():
+    dummy_hits = {"hits": []}
+    mocked_response = Response(
+        search="", response={"test": "hello", "aggregations": {}, "hits": dummy_hits}
+    )
+    mocked_response._search = MagicMock()
+    mocked_response._d_ = dummy_hits
+    mocked_response._hits = MagicMock()
+    mocked_response.hits = MagicMock()
+    mocked_response.hits.total.value = 0
+    output = score(mocked_response)
+
+    expected_score = {"total": len(dummy_hits["hits"])}
+    assert output == expected_score
+
+
+def test_score_with_hits():
+    mocked_response = Response(
+        search="",
+        response={
+            "took": 19,
+            "timed_out": False,
+            "_shards": {"total": 1, "successful": 1, "skipped": 0, "failed": 0},
+            "hits": {
+                "total": {"value": 2326, "relation": "eq"},
+                "max_score": 0.0,
+                "hits": [],
+            },
+            "aggregations": {
+                "missing_edu_context": {"doc_count": 3},
+                "missing_object_type": {"doc_count": 5},
+                "missing_description": {"doc_count": 818},
+                "missing_license": {"doc_count": 5},
+                "missing_title": {"doc_count": 0},
+                "missing_keywords": {"doc_count": 1},
+                "missing_ads_qualifier": {"doc_count": 1628},
+                "missing_subjects": {"doc_count": 1},
+                "missing_material_type": {"doc_count": 12},
+            },
+        },
+    )
+    mocked_response._search = MagicMock()
+    mocked_response._d_ = {"hits": []}
+    mocked_response._hits = MagicMock()
+    mocked_response.hits = MagicMock()
+    mocked_response.hits.total.value = 0
+    output = score(mocked_response)
+
+    expected_score = {
```

Comment:
Should we add a `BaseModel` for this structure? Or is the set of keys somehow dependent on the query? If so, we could still eventually declare a `BaseModel` where all values would be optional (and omit missing values in the response serialization)

Reply:
Mmmh, since this is a test, I would argue against it. I prefer tests to be as explicit as possible, i.e., depend on little else. In this case, if I change something elsewhere a little, my test fails earlier - which I find a good thing. What do you think?

Reply:
Yes, I was hoping, that the endpoints might work with returning baseclass structures (see other comment) - If that would have beend the case and serialization (baseclass instance->json) would be done automagically by fastapi, then we could have written the tests on top of the baseclass types.

I agree to having as indepdendent and explicit tests as possible!

But I also always want to reduce the use of dictionaries for scnearios like this (where they have to have some implied structure assumed by the consumer of the dictionary).

Reply:
Good point. Do you suggest to use better return structures, e.g., models, for those functions for which we have tests, etc.?